tests/cupy_tests/core_tests/test_array_function.py

```diff
@@ -28,3 +28,16 @@ def test_array_function(self):
         else:
             self.assertEqual(qr_cpu.dtype, qr_gpu.dtype)
             cupy.testing.assert_allclose(qr_cpu, qr_gpu, atol=1e-4)
+
+    @testing.numpy_cupy_equal()
+    def test_array_function_can_cast(self, xp):
```

Comment:
Could you add `@testing.with_requires('numpy>=1.17')`?

Reply:
I think it would be best not to add a NumPy version requirement, after all, we want the implementation in this PR to work with any version.

Reply:
`__array_function__` does not work by default until in NumPy 1.16, so this test seems not to work. Could you use `cupy.can_cast` instead of `numpy.can_cast`?

Reply:
You're right, here we can only test with NumPy>= 1.17. But we also want to test the behavior with `__array_function__`, that's why we need to use `numpy.can_cast`. What I will do is add a different test that runs independent of NumPy version, but uses `cupy.can_cast` then.